File: back-end/back-end/src/main/java/com/datn/backend/model/hoa_don/HoaDonTraHang.java
package com.datn.backend.model.hoa_don;

import com.datn.backend.model.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HoaDonTraHang extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String ma;
    private String tenNguoiNhan;
    private String sdtNguoiNhan;
    private String emailNguoiNhan;
    private String diaChiNguoiNhan;
    private BigDecimal tongTien;
    private BigDecimal tongTienPhieuGiamGiaCu;
    private BigDecimal tongTienPhieuGiamGiaMoi;
    private BigDecimal tongTienTraKhach;
    private String ghiChu;

    @OneToMany(mappedBy = "hoaDonTraHang", cascade = CascadeType.ALL)
    private List<HoaDonChiTiet> hoaDonChiTiets;

    @OneToOne
    @JoinColumn(referencedColumnName = "id")
    private HoaDon hoaDon;
}

File: admin-fe/src/app/model/class/hoa-don-tra-hang.ts
import { HoaDonChiTiet } from "./hoa-don-chi-tiet.class";
import { HoaDon } from "./hoa-don.class";

export class HoaDonTraHang {
  ma: string;
  tenNguoiNhan: string;
  sdtNguoiNhan: string;
  emailNguoiNhan: string;
  diaChiNguoiNhan: string;
  tongTien: number;
  tongTienPhieuGiamGiaCu: number;
  tongTienPhieuGiamGiaMoi: number;
  tongTienTraKhach: number;
  ghiChu: string;
  hoaDonChiTiets: HoaDonChiTiet[];
  hoaDon: HoaDon;

  createdAt: string;
  createdBy: string;
  updatedAt: string;
  lastUpdatedBy: string;
}
